src/Test/NodeTestCase.php

```diff
@@ -24,11 +26,23 @@ abstract class NodeTestCase extends TestCase
      */
     private $currentEnv;
 
-    abstract public function getTests();
+    public function getTests()
+    {
+        return [];
+    }
+
+    public static function provideTests(): iterable
```

Comment:
I suggest adding a phpdoc `@return iterable<array{0: Node, 1: string, 2?: Environment|null, 3?: bool}>` to document the expected signature of the data provider (based on the test signature)

Reply:
Done, thanks.